src/android/CameraLauncher.java

```diff
@@ -778,7 +778,22 @@ private void processResultFromGallery(int destType, Intent intent) {
 
                     // If sending base64 image back
                     if (destType == DATA_URL) {
-                        this.processPicture(bitmap, this.encodingType);
+                        // To allow keeping the Exif data, we need to save the image to allow ExifInterface add the data
+                        // After that, we load the image and send it as base64
+                        try {
+                            String modifiedPath = this.outputModifiedBitmap(bitmap, uri, mimeTypeOfGalleryFile);
+                            InputStream fileStream = FileHelper.getInputStreamFromUriString(modifiedPath, cordova);
+                            byte[] file = new byte[fileStream.available()];
```

Comment:
Using `available()` here is unsafe. `available` API is used to determine how many bytes are available to read without blocking the current thread. It may return `0` or any number that is less than the actual size of the file. This can lead to file corruption since the byte buffer is used to attempt to read the entire file. See https://developer.android.com/reference/java/io/InputStream#available()

Instead, create a chunk sized buffer and use the [read](https://developer.android.com/reference/java/io/InputStream#read(byte[],%20int,%20int)) API in a loop until `read` returns `-1` which tells you that you read the entire file. Note that the whole buffer may not be filled, so using the return value to know how many bytes were read is required.

Reply:
working on it !